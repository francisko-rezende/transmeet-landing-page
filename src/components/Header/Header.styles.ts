import styled from 'styled-components'

export const Bg = styled.div`
  position: sticky;
  z-index: 4;
  top: 0;
  background-color: ${({ theme }) => theme.colors.typographyColors.black};
`

export const Header = styled.header`
  padding: 4.1rem 0 3.4rem 0;
  display: flex;
  align-items: center;
  justify-content: space-between;
`

export const OnlyOnSmallerScreens = styled.div`
  display: block;
  @media (min-width: 808px) {
    display: none;
  }
`

export const OnlyOnLargerScreens = styled.div`
  display: block;
  @media (max-width: 808px) {
    display: none;
  }
`
