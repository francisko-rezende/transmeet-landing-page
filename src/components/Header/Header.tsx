import React from 'react'

import AppLinks from 'components/AppLinks'
import MainContainer from 'components/MainContainer'
import MobileMenu from 'components/MobileMenu'
import NavBar from 'components/NavBar'
import Image from 'next/image'

import * as S from './Header.styles'

const Header = () => {
  return (
    <S.Bg>
      <MainContainer>
        <S.OnlyOnLargerScreens>
          <S.Header>
            <Image
              src="/images/logo-color.webp"
              width={200}
              height={45}
              alt="transmeet"
            />
            <NavBar />
            <AppLinks />
          </S.Header>
        </S.OnlyOnLargerScreens>
        <S.OnlyOnSmallerScreens>
          <MobileMenu />
        </S.OnlyOnSmallerScreens>
      </MainContainer>
    </S.Bg>
  )
}

export default Header
