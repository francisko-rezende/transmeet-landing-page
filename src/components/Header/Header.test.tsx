import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Header from '.'

describe('<Header />', () => {
  test('should render the header', () => {
    const { container } = renderWithTheme(<Header />)

    expect(screen.getByRole('banner')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
