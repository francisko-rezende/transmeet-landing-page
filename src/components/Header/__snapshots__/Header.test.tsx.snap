// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render the header 1`] = `
.c10 {
  padding: 1.4rem 4.4rem;
  border-radius: 24px;
  background-color: #BB4981;
  color: #FFFFFF;
  font-size: 16px;
  font-weight: 700;
  cursor: pointer;
}

.c10:hover {
  background-color: #8B426E;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 4.2rem;
}

.c9 {
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
  width: 100%;
  margin: 0 auto;
  font-size: 16px;
}

.c9:hover {
  font-weight: 700;
}

.c9:hover::after {
  content: '';
  position: absolute;
  display: block;
  bottom: -1rem;
  height: 0.5rem;
  background: linear-gradient( 269.71deg, rgba(255,255,255,0) 0.01%, #f7c2d2 19.66%, #bb4981 49%, #185aa1 74.55%, #1f478b 100% );
  -webkit-animation: hoverAnimation 0.3s forwards;
  animation: hoverAnimation 0.3s forwards;
}

.c1 {
  max-width: calc(1170px + 44px);
  padding: 0 22px;
  margin: 0 auto;
}

.c17 {
  background-color: transparent;
  border: none;
  cursor: pointer;
  width: 17px;
  height: 17px;
}

.c15 {
  background-color: transparent;
  border: none;
  cursor: pointer;
  width: 22px;
  height: 14px;
}

.c12 {
  display: grid;
  grid-template-columns: repeat(3,1fr);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  justify-items: end;
  padding: 16px 0;
}

.c13 {
  grid-column: 2/3;
  place-self: center;
}

.c14 {
  grid-column: 3/4;
}

.c20 {
  margin-top: 110px;
  margin-bottom: 108px;
}

.c19 {
  list-style: none;
}

.c22 {
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
  width: 100%;
  margin: 0 auto;
  font-size: 16px;
}

.c22:hover {
  font-weight: 700;
}

.c22:hover::after {
  content: '';
  position: absolute;
  display: block;
  bottom: -1rem;
  height: 0.5rem;
  background: linear-gradient( 269.71deg, rgba(255,255,255,0) 0.01%, #f7c2d2 19.66%, #bb4981 49%, #185aa1 74.55%, #1f478b 100% );
  -webkit-animation: gubmQU 0.3s forwards;
  animation: gubmQU 0.3s forwards;
}

.c16 {
  opacity: 0;
  background: #152536;
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  height: 100vh;
  overflow: hidden;
  pointer-events: none;
  -webkit-transition: opacity 0.3s ease-in-out;
  transition: opacity 0.3s ease-in-out;
}

.c16 > button[aria-label='Fechar menu'] {
  position: absolute;
  top: 0;
  right: 0;
  margin: 20px 50px;
}

.c16 .c18 > :last-child {
  padding-top: 100px;
}

.c16 .c18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c16 .c21 {
  color: #FFFFFF;
  color: #FFFFFF;
}

.c16 .c21:hover::after {
  content: unset;
}

.c4 {
  max-width: 40rem;
}

.c5 {
  display: grid;
  grid-template-columns: repeat(4,minmax(0,1fr));
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  list-style: none;
}

.c6 {
  width: 10rem;
  text-align: center;
}

.c7 {
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
  width: 100%;
  margin: 0 auto;
  font-size: 16px;
}

.c7:hover {
  font-weight: 700;
}

.c7:hover::after {
  content: '';
  position: absolute;
  display: block;
  bottom: -1rem;
  height: 0.5rem;
  background: linear-gradient( 269.71deg, rgba(255,255,255,0) 0.01%, #f7c2d2 19.66%, #bb4981 49%, #185aa1 74.55%, #1f478b 100% );
  -webkit-animation: hoverAnimation 0.3s forwards;
  animation: hoverAnimation 0.3s forwards;
}

.c0 {
  position: -webkit-sticky;
  position: sticky;
  z-index: 4;
  top: 0;
  background-color: #152536;
}

.c3 {
  padding: 4.1rem 0 3.4rem 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c11 {
  display: block;
}

.c2 {
  display: block;
}

@media (min-width:811px) {
  .c11 {
    display: none;
  }
}

@media (max-width:811px) {
  .c2 {
    display: none;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <header
        class="c3"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27200%27%20height=%2745%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="transmeet"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
        <nav
          class="c4"
        >
          <ul
            class="c5"
          >
            <li
              class="c6"
            >
              <a
                class="c7"
                href="/#"
              >
                Home
              </a>
            </li>
            <li
              class="c6"
            >
              <a
                class="c7"
                href="/"
              >
                Aplicativo
              </a>
            </li>
            <li
              class="c6"
            >
              <a
                class="c7"
                href="/about"
              >
                Sobre
              </a>
            </li>
            <li
              class="c6"
            >
              <a
                class="c7"
                href="/#"
              >
                Contato
              </a>
            </li>
          </ul>
        </nav>
        <div
          class="c8"
        >
          <div>
            <a
              class="c9"
              href="#"
            >
              Entre
            </a>
          </div>
          <a
            class="c10"
          >
            Cadastrar
          </a>
        </div>
      </header>
    </div>
    <div
      class="c11"
    >
      <menu
        class="c12"
      >
        <div
          class="c13"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27117%27%20height=%2726%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="transmeet"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </div>
        <div
          class="c14"
        >
          <button
            aria-label="Abrir menu"
            class="c15"
            height="14px"
            width="22px"
          >
            <svg
              fill="none"
              viewBox="0 0 22 14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M20.634 13.804H1.145c-.633 0-1.145-.359-1.145-.802v-.804c0-.444.512-.803 1.145-.803h19.49c.632 0 1.147.36 1.147.803v.804c0 .443-.515.802-1.148.802Zm0-5.62H1.145C.512 8.184 0 7.824 0 7.381v-.802c0-.443.512-.803 1.145-.803h19.49c.632 0 1.147.36 1.147.803v.802c0 .444-.515.803-1.148.803Zm0-5.619H1.145C.512 2.565 0 2.205 0 1.762V.959C0 .516.512.157 1.145.157h19.49c.632 0 1.147.36 1.147.802v.803c0 .443-.515.803-1.148.803Z"
                fill="#fff"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
        <nav
          aria-hidden="true"
          class="c16"
        >
          <button
            aria-label="Fechar menu"
            class="c17"
            height="17px"
            width="17px"
          >
            <svg
              fill="none"
              viewBox="0 0 18 18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M14.916 17.152 1.136 3.37C.687 2.923.582 2.303.901 1.985l.575-.576c.318-.318.938-.213 1.386.234l13.78 13.781c.448.448.554 1.07.236 1.387l-.575.576c-.319.318-.94.212-1.388-.235Z"
                fill="#fff"
                fill-rule="evenodd"
              />
              <path
                clip-rule="evenodd"
                d="M2.866 17.152 16.646 3.37c.448-.448.552-1.068.234-1.386l-.575-.576c-.318-.318-.938-.213-1.386.234L1.14 15.424c-.448.448-.554 1.07-.236 1.387l.575.576c.318.318.94.212 1.388-.235Z"
                fill="#fff"
                fill-rule="evenodd"
              />
            </svg>
          </button>
          <ul
            class="c18 c19"
          >
            <div
              class="c20"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27196%27%20height=%2743%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="transmeet"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </div>
            <li
              class=""
            >
              <a
                class="c21 c22"
                href="/#"
              >
                Home
              </a>
            </li>
            <li
              class=""
            >
              <a
                class="c21 c22"
                href="/#"
              >
                Aplicativo
              </a>
            </li>
            <li
              class=""
            >
              <a
                class="c21 c22"
                href="/#"
              >
                Sobre
              </a>
            </li>
            <li
              class=""
            >
              <a
                class="c21 c22"
                href="/#"
              >
                Contato
              </a>
            </li>
            <li
              class=""
            >
              <a
                class="c10"
              >
                Entrar/Registrar
              </a>
            </li>
          </ul>
        </nav>
      </menu>
    </div>
  </div>
</div>
`
